src/components/Modal/src/ModalLayer.vue

```diff
@@ -175,7 +175,11 @@ export default {
 		closeOnClickOutside(event) {
 			const { closeOnClickOutside } = this.currentLayer.state.options;
 			const { modal } = this.$refs;
-			if (modal && closeOnClickOutside && !modal.contains(event.target)) {
+			if (
+				modal
+				&& closeOnClickOutside
+				&& modal.compareDocumentPosition(event.target) === 10
```

Comment:
```suggestion
				&& modal.compareDocumentPosition(event.target) === (Node.DOCUMENT_POSITION_PRECEDING + Node. DOCUMENT_POSITION_CONTAINS)
```

Hmm, I thought we had [`no-magic-numbers`](https://eslint.org/docs/rules/no-magic-numbers) enabled.